Unprepared documentation (#286)
scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver: docs/src/unprepared_statements.md

# Unprepared statements

> Whenever you don't want to execute prepared statements, you need to inform the driver what the inserted data is.

Reviewer: ```suggestion
Whenever you want to execute statements without their prior preparation, you need to inform the driver what the inserted data is.
```

There are multiple ways you can achieve that:

1. passing recognizable types,

> 2. passing the value, alongside the type hint,

Reviewer: ```suggestion
2. passing a _type hint_ alongside the value,
```

on how info field should look like.

#### Examples

```javascript
// equivalent to `int`
const intHint = { code: 0x9 /* types.dataTypes.int */ }; 

// equivalent to `map<text, boolean>`
const mapHint = { 
  code: 0x21 /* types.dataTypes.map */, 
  info: [
    { code: 0xa /* types.dataTypes.text */ },
    { code: 0x4 /* types.dataTypes.boolean */ }
  ] 
};

// equivalent to `vector<float, 3>`
const vectorHint = { 
  code: 0x0 /* types.dataTypes.custom */, 
  customTypeName: "vector",
  info: [{ code: 0x8 /* types.dataTypes.float */ }, 3] 
};
```

____

Hints can be provided for all, or only some of the parameters.
A hint can be full, or partial. A hint is considered full when
it specifies either a simple type (like int or date) or a complex type (like a list or a map) with
all of its subtypes specified. A hint is considered partial if it defines a complex type without
specifying hints for its subtypes. Providing hints for only some subtypes — such as specifying
the type of map keys but not the values — is not considered a valid hint.
When a partial hint is provided, the driver attempts to guess the remaining part of the type
of that value.

> In prepared statements any hints you provide will be ignored.

Reviewer: ```suggestion
Hints can be provided for all or only some of the parameters.
A hint can be full or partial. A hint is considered _full_ when
it specifies either:
- a simple type (like int or date),
- or a complex type (like a list or a map) with all its subtypes specified.
A hint is considered _partial_ if it defines a complex type without
specifying hints for its subtypes.
A complex type hint with hints provided for only some subtypes — such as specifying
the type of map keys but not the values — is not considered a valid hint.
When a partial hint is provided, the driver attempts to guess the remaining part of the value's type.

In prepared statements any hints you provide will be ignored.
```
